lib/date_range.rb

```diff
@@ -0,0 +1,23 @@
+require 'date'
+
+module Hotel
+  class DateRange
+    attr_reader :check_in, :check_out, :nights_spent
+
+    def initialize(check_in, check_out)
+      raise ArgumentError, 'Invalid date range given.' unless check_out > check_in
```

Comment:
It's helpful to include the invalid arguments when you raise an `ArgumentError`.

```suggestion
      raise ArgumentError, "Invalid date range given. check_in: #{check_in} check_out: #{check_out}" unless check_out > check_in
```